### ISSUE TITLE
Unable to execute named parameters using '@' in the parameter

### ISSUE THREAD

**Issue by chaixshot** [overextended/oxmysql#48]
Version 1.6.0 at same execute fucntion
```
MySQL.ready(function()
	MySQL.Async.execute('INSERT INTO discord(identifier, discord) VALUES(@identifier, @discord)',
	{
		['@identifier'] = "steam:11000013d8d70d5",
		['@discord'] = "xbl:2535406267274231",
	})
end)
```
![image](https://user-images.githubusercontent.com/13221018/137578680-0496ea31-d9f3-467e-b037-a9543a4dbd7b.png)

**Comment by darksaid98** [comment]
The column `discord` in table `discord` of your database can not be set to `NULL`, since you haven't enabled it for said column.

Not an oxmysql issue.

**Comment by chaixshot** [comment]
![image](https://user-images.githubusercontent.com/13221018/137583527-b1099712-fff9-4a55-9a05-c2c44f167d8a.png)
Look like mysql-async syntax have problem.

**Comment by darksaid98** [comment]
It should be `['identifier'] = 'test1'` not `['@identifier'] = ...`.

I could have sworn there was backward compatibility for putting '@' in there but I might be wrong, so let me know how that goes.

**Comment by chaixshot** [comment]
![image](https://user-images.githubusercontent.com/13221018/137584112-f7087ed2-487b-40fb-8922-7692fc9db1e8.png)

This form 1.5.1 (Work)
![image](https://user-images.githubusercontent.com/13221018/137584326-d29681ff-defe-480f-ad66-0006ff84fbbe.png)


 This form 1.6.0 (Not work)
![image](https://user-images.githubusercontent.com/13221018/137584340-106d2028-de99-4356-af14-552ec44a6448.png)


Look like mysql-async syntax compatible not working.
Because this is error on all my old script just update 1.5.1 to 1.6.0

```
MySQL.ready(function()
	local test1 = MySQL.Sync.fetchAll('SELECT id FROM users WHERE identifier = ?',
	{
		"steam:11000010971396e"
	})
	print(test1[1])
	
	---------------------------
	
	local test2 = MySQL.Sync.fetchAll('SELECT id FROM users WHERE identifier = @identifier',
	{
		['@identifier'] = "steam:11000010971396e"
	})
	print(test2[1])
	
	local test3 = MySQL.Sync.fetchAll('SELECT id FROM users WHERE identifier = @identifier',
	{
		['identifier'] = "steam:11000010971396e"
	})
	print(test3[1])
end)
```

**Comment by darksaid98** [comment]
Yea that's what I thought. Seems there's been a regression in backward compatibility in 1.6. 

I'm currently digging through the codebase patching another issue. I'll check later to see if this is trivial to re-introduce.

**Comment by chaixshot** [comment]
![image](https://user-images.githubusercontent.com/13221018/137584615-950f17c7-2687-4e3c-82a7-8ca9ef0497d9.png)

Oh one thing update support for nil parameters doesn't seem to work
![image](https://user-images.githubusercontent.com/13221018/137584633-7ff85634-6215-4ec5-9472-76a76bf6c68b.png)

**Comment by darksaid98** [comment]
This is the same issue as in your first post. Again, the issue is your database setup. You need to allow `discord` to be set to `NULL`.

**Comment by chaixshot** [comment]
> Yea that's what I thought. Seems there's been a regression in backward compatibility in 1.6.
> 
> I'm currently digging through the codebase patching another issue. I'll check later to see if this is trivial to re-introduce.

Ok, for now i will try this.
```
local function safeArgs(query, parameters, cb, transaction)
	if type(query) == 'number' then query = Store[query] end
	if transaction then
		assert(type(query) == 'table', ('A table was expected for the transaction, but instead received %s'):format(query))
	else
		assert(type(query) == 'string', ('A string was expected for the query, but instead received %s'):format(query))
	end
	if cb then
		assert(type(cb) == 'function', ('A callback function was expected, but instead received %s'):format(cb))
	end
	local type = parameters and type(parameters)
	if type and type ~= 'table' and type ~= 'function' then
		assert(nil, ('A %s was expected, but instead received %s'):format(cb and 'table' or 'function', parameters))
	end
	
	local nparameters = parameters
	if parameters then
		nparameters = {}
		for name, value in pairs(parameters) do
			nparameters[string.gsub(name, "@", "")] = value
		end
	end
	
	return query, nparameters, cb
end
```

**Comment by thelindat** [comment]
> 
> 
> Yea that's what I thought. Seems there's been a regression in backward compatibility in 1.6.
> 
> I'm currently digging through the codebase patching another issue. I'll check later to see if this is trivial to re-introduce.

Can you confirm that both [':var'] and ['@var'] do not work? I'll have to look back to see what I've changed to break this (likely in the patch for named-parameters).

I'm on New World for now but if you don't find a solution before me then I'll take a look at it.

**Comment by chaixshot** [comment]
[':var'] and ['@var'] not working both.
![image](https://user-images.githubusercontent.com/13221018/137586532-6d132b0a-011c-4b80-b68d-22d364b533d9.png)

**Comment by chaixshot** [comment]
![image](https://user-images.githubusercontent.com/13221018/137586631-e8f1356a-1b4b-46bf-bd27-601957cd67ba.png)

**Comment by darksaid98** [comment]
Fixed in #50.
